### PR TITLE
[DNM] Grues

### PR DESCRIPTION
**Pull request by Shadowmech88**
Grues deal 15 brute damage per hit, can't be stunned, and move at twice the speed of humans, but they take double damage from both brute and burn.
Grues also have inverted vision, meaning they see light as dark, and dark as light. This means that they can see perfectly in a pitch-black room, but can't see anything in a well-lit room.
image image

Grues have a spell they can use to create a negative light source centered on themselves, swallowing all light around them for up to a minute.
Grues can heal themselves by butchering humans with their claws and consuming human meat, 50 of each damage type is healed when they eat it.

I figure down the line they could make an interesting antag or random event or something.

This PR also adds a new charge type for spells that allows their charge to be drained gradually while the spell is active, and stop draining when the spell is deactivated, which happens when the spell runs out of charge, or can be done at any time by clicking the spell icon again.
It also makes humans butcherable. They still can't be gibbed via butchering, but you can get meat and teeth from them now. I'm not sure why that was disabled.

### CODE REVIEW

**Comment by SkowronX** [comment]
gsgrue

**Comment by Kurfursten** [comment]
When you butcher a human, it should add to their description text:

> \He is likely to have been eaten by a grue.

**Comment by SkowronX** [comment]
That violation of present perfect tense, though.

**Comment by jknpj** [comment]
@Kurfursten That description is way too **Gruesome**

naamloos-2

**Comment by Shadowmech88** [comment]
Should I go with
"He seems to gave been eaten by a grue." or
"It is likely he was eaten by a grue."?

**Comment by carshalash** [comment]
Oh boy! Features!

**Comment by AI-Is-Rogue** [comment]
> "It is likely he was eaten by a grue."

Seems more eerie and mysterious, I'd go with that.

**Comment by FudgePucker** [comment]
Oh man this seems like a fun side antag that can spawn in mixed.
Maintenance will be scarier if there are grues walking around in them.

Maybe a gimmicky feature you can add to them is some type of innate ability to unbolt airlocks, shock resistance, and always has maintenance access, so that way they can move around the station with out AI just bolting them in someplace.

EDIT: Another idea for a power would be flash of insight. It gives you xray for ten seconds. This would allow you to see other dark areas to hide in if you need to move though a main hallway to get to another maintenance tunnel. Pretty much you activate it, then run for the next maintenance. Maybe also make them invisible for the duration but unable to do anything but open airlocks, acting like a false phase shift, letting them run though areas without getting caught as fast, but still can bump into people and leave a trail of airlocks.
Imagine how pants shitting-ly scary it would be to see airlocks opening randomly next to you and feeling the presence of a beast pass by you as it runs into the nearest dark room.

**Comment by Shadowmech88** [comment]
> airlocks

I didn't think about that. They'd need an assistant ID just to run between maintnance shafts.

I'm inclined to give them an ability to force open _any_ airlock, powerless, bolted or otherwise. That way they can't be stopped or trapped by an AI.

As far as running between maintenance halls, grues can already use their swallow darkness ability to do that. It lets them see where they're going, and to everyone else it just looks like a big blob of darkness moving through the hallway.

**Comment by Kurfursten** [comment]
@SkowronX The awkward grammar is itself a reference to "You are likely to be eaten by a grue"

**Comment by FudgePucker** [comment]
@Shadowmech88 
How would they get an assistant ID? Ask the HoP for one?

The glob of darkness would be a dead giveaway.
Although it would be pretty interesting in a metaphorical way, as in the dark maintenance he can see everything while they only see darkness, and he sees them as blobs of darkness as well, so having it be the reverse would be pretty neato. A unique dynamic where the situation is reversed depending on if he is in maintenance or not. The main hallways being the grues maintenance.
But in a practical sense it is dumb. Blob of darkness just ran into that maintenance room? VALIDS SECURE, LETS GO GET HIM BOYS, as the entire station sweeps that maintenance area. If he is going to have that dead of a give away to move though the station, then he needs some more fire power, wizards can be spotted right away, but they know magic, nuke ops can be spotted right away, that's why they have an arsenal. The Grue doesn't get any stunning assault rifles or any spells, if he is spotted going  in a maintenance tunnel he is fucked, doubly so because he had to use some of his cooldown to move there. Our server is too valid hungry to let some valid survive like that.

**Comment by Shadowmech88** [comment]
I could make them hit harder. As it is going after one alone is a death sentence, because their darkness renders flashlights useless and they have a high chance of knocking you down due to their hulk, but I didn't account for 10 people all swarming the same maintenance shaft.

I could also give them a longer-cooldown spell to actually blow out light sources, so they could easily increase their territory beyond just maintenance. That'd make it more difficult to track their movements.

**Comment by Shadowmech88** [comment]
Also I should note that their speed would make them difficult to hit with projectiles, and flat-out impossible to chase.

**Comment by Kurfursten** [comment]
What's the range on the negative light spell, and the cooldown?

If the AOE dark range is pretty short, then I see no problem with them ripping open any airlock, since fighting in hallways would still be extremely risky for the grue. If the range is fairly large, then they don't need help from airlocks - an abomination that horrible, swallowing whole hallways into darkness, should be something that someone has to blunder into unleashing.

**Comment by Shadowmech88** [comment]
> What's the range on the negative light spell, and the cooldown?

It fills nearly the entire screen with darkness. It doesn't have a cooldown _per se_. It has a max charge of 60 seconds. While active, the charge drains. The ability automatically deactivates when the charge hits 0, or can be manually deactivated by clicking the spell again. While inactive, the charge increases until it reaches 60 again.

I'd still like them to be able to force open airlocks, so that they can navigate by themselves.  I feel that the darkness is small enough to give enemies a rough picture of where the grue is within is, and they're fragile enough that someone firing wildly into the approaching darkness has a good chance of hitting the grue if they aim for its center.

The main purpose of the spell was to provide the grue with sight and a small amount of protection while traversing lit areas, and render flashlights useless while it hunts in dark areas.

**Comment by Kurfursten** [comment]
I'm not sure they need help with airlocks if they can fill almost an entire screen with darkness. That means they actually fight better out in the open of a hallway than in maintenance, since they will both have a wider field of view, obscure more tiles from the enemy, and have more room to maneuver. No reason to retreat to maint except to recharge, and you can do that just as easily in a closet or disposal bin.

In maintenance the darkness doesn't even really matter because it's 2 wide at most and frequently 1 wide, so you can just spam lasers down. All of this sounds like it's adding up to Grues being more of a brutal frontal assault main-hallways type of foe than a stealthy lurk in maintenance foe.

At 15 brute, no one's dying without screaming the location of the grue, but at the same time, a grue is actually at his weakest while in the maint due to burn vulnerability and narrow tunnels.

**Comment by Shadowmech88** [comment]
The darkness fills almost an entire screen, but only if you're standing right on top of it. Otherwise it's a pretty distinct moving mass, the center of which enemies only need to pepper with shots to have a good chance of hitting it.

Whereas in Maintenance, enemies have no frame of reference as to the grue's position, no moving target to hit. He could be running circles around them and they wouldn't even know it, doubly so for open areas in which the grue has destroyed light sources.

Maintenance might not be super ideal for a fight, but I think a grue would be safer there than, for example, trying to fight armed opponents in a well-lit bar, darkness spell or not.

**Comment by Shadowmech88** [comment]
I think I will add a spell that lets grues destroy light fixtures, to make it easier for them to expand their territory out of maintenance.

**Comment by Kurfursten** [comment]
I'm not sure "territory" is necessarily an important concept for a Grue of your design. The way you've built them -- very vulnerable, squishy creatures that need to quickly murder lots of people so they have a source of regeneration meat before the lasers come out -- is more akin to a rapid brawler. That is, more like nook ops than aliums or a blab. Your grue benefits from staying on the move because once he's found out, all it takes is science printing one pair of NVGs and he's pretty much done. Double damage means he's getting his face smashed in if anyone is still alive.

**Comment by ghost** [comment]
Grues are known for seldom being seen, they should be geared towards stealth, not expansion; and they should have mechanisms to prevent any assistant from killing it and displaying it for everyone to see.
What I propose, as per is that light doesn't just blind them, but harm and eventually ash them.

They should also be able to get on all fours to go even faster, hide, and go under people's feet, like mouses; but also being unable to harm people. A nice alternative when the spell is drained so it has less charge can be less commonly used, for the critical "kill someone"/"escape"/"cross past strong lights" moments. 
Like, shit, imagine traversing maint with just you PDA's light and you almost miss a shadow zipping past you without even being sure what it was.

EDIT: I'm stupid, indefinitely buffing the speed of a fast critter is a bad idea. Maybe making it a "sprint" and tie it to a charge "spell", like the anti-light?

**Comment by Shadowmech88** [comment]
> Your grue benefits from staying on the move 

But only staying on the move within darkness. Otherwise they can be clearly seen, tracked, and shot at.
Before the NVGs come out, the grue would benefit from increasing the number of dark areas on the station, since it would allow him to move and hide without giving away his position.

**Review by PJB3005 — APPROVED** [review]

**Comment by Shadowmech88** [comment]
There are at least a few changes yet to be made.

**Comment by SonixApache** [comment]
on a related note: box maint SUCKS

**Comment by Exxion** [comment]
We need maint more like the original Efficiency Station (what Deff is based on, for anyone who doesn't know)
Personally, I like Deff's massive, sprawling "maint", but I understand how it would be too much for many people.
But the original Efficiency Station's maint is legendary. It's not gimmicky like Deff's or anything, it's just extremely well-executed maint.

That station is really the only thing I miss from when I used to play /tg/, long ago
Well, that and their hyper-advanced janitech

**Comment by icantthinkofanameritenow** [comment]
I guess this could work as a random event spawned pseudo-antagonist. We'll just have to keep it from being too similar to the shadowlings that other servers have.

**Comment by gbasood** [comment]
> grues

are we lifeweb
it's fine for bus I guess though

**Comment by Zennyization** [comment]
If I was playing a Grue, I probably would not hurt anybody for a long time. I'd take pleasure in following them around without them knowing I was there. It would be even funnier if they were able to create noises such as the "spooky" noise from the Hidden that is in the sound files already.

**Comment by FudgePucker** [comment]
Over powered idea, immunity to thermals and NVGs.
If they are in light they can still be seen, but if they are in the dark then enhanced vision doesn't work.

It would make Grue vs Aliums very fun to watch.
Also would spook the shit out of people using NVGs, 'oh man I can see anything come up to me in maintenance, nothings going to get the drop on me- OH SHIT FUCK DICKS AAAAAAAAAAA'

**Comment by Shadowmech88** [comment]
Also in case anyone's curious, the color matrix for grues is a property of their eyes, meaning if you kill one, you can remove its eyes and have someone transplant them into you to gain its sight.

**Comment by Kurfursten** [comment]
We need an umbrella item that creates darkness in 1x1 area, like a reverse PDA light.

dark shadows umbrella

File: code/modules/mob/living/carbon/racial_spells.dm [hunk 1]
```diff
@@ -0,0 +1,35 @@
+//For spells designed to be inherent abilities given to mobs via their species datum
+
+/spell/racial
```
Comment (ComicIronic) [inline code/modules/mob/living/carbon/racial_spells.dm:3]: What's the point of this type? Racial spells have no reason to be grouped together, and you're restricting them from being aoe_turf or targeted.

Comment (Shadowmech88) [reply code/modules/mob/living/carbon/racial_spells.dm:3]: I figured it'd be better than setting the `panel` and `override_base` vars on each new racial spell, but I can remove the subtype if you think that would be better.

Comment (ComicIronic) [reply code/modules/mob/living/carbon/racial_spells.dm:3]: I'd rather people were made to do that small amount of manual work over someone seeing this subtype and deciding all racial spells should go under it before promptly copying the entirety of `spell/targeted` into a new type.

**Comment by clusterfack** [comment]
I do think there should be some sort of /self spell subtype for spells like hide and a few others which have no requirements and no targets.

**Comment by Tacolizard** [comment]
grues shouldn't be able to attack anybody in the light. they should be only capable of causing crewmembers to creepily disappear, and should NEVER be seen.

**Comment by Tacolizard** [comment]
also they should be very a much a side antag. they should not be geared towards creating darkness and antagonizing the crew, but just roaming in the dark to pick off people.

**Comment by Gunosaguki** [comment]
Bring this in already!

**Comment by kol1th** [comment]
...When?

**Comment by Shadowmech88** [comment]
Closing in favor of #12848 due to DMCA fuckery.

**Pull request closed without merging**